### PR TITLE
Add support for CTAs in EmbeddedEntryBlock

### PR DESCRIPTION
**Pull request by hinzed1127**
Jira ticket: [LDAF-377](https://ldaf.atlassian.net/browse/LDAF-377)

## Proposed changes

- Adds Call To Action as a content type that can be embedded in the rich text descriptions for core content pages ([ref in contentful](https://app.contentful.com/spaces/pc5e1rlgfrov/content_types/serviceGroup/fields)
- Fetches that extra content in the GraphQL query and renders accordingly

## Acceptance criteria validation

- [x] CTAs render in core content description fields

## Other details

[Here's an example page where I've added some CTA embeds to the description (in a draft state)](https://ldaf-okjxgf55d-ldaf.vercel.app/land/fire/safety)

[LDAF-377]: https://ldaf.atlassian.net/browse/LDAF-377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts [hunk 1]
```diff
@@ -305,7 +313,7 @@ export const load = async ({
     if (!pageMetadata) break fetchData;
     const client = getContentfulClient({
       spaceID: CONTENTFUL_SPACE_ID,
-      token: CONTENTFUL_DELIVERY_API_TOKEN,
+      token: CONTENTFUL_PREVIEW_API_TOKEN,
```
Comment (hinzed1127) [inline src/routes/(infoPages)/[topTierPage]/[...serviceGroupPage]/+page.server.ts:316]: Added this in to work on developing the feature/displaying it on the test site. will remove this and the `preview:true` flags before merging

**Review by hinzed1127 — COMMENTED** [review]
Self-review

File: src/lib/fragments/entryProps.ts [hunk 2]
```diff
@@ -41,5 +41,17 @@ export default gql`
...
+        json
+        links {
+          assets {
+            hyperlink {
+              ...AssetProps
+            }
+          }
+        }
+      }
+    }
```
Comment (hinzed1127) [inline src/lib/fragments/entryProps.ts:55]: As this has grown, I wonder if it's worth splitting this into separate fragments for each of the entry types (`ImageWrapper`, `Contact`, and `CallToAction`)? It seems like adding some unnecessary blot to queries and it'd be more explicit what we're expecting where if we broke it up

**Review by LouisFettet — COMMENTED** [review]
LGTM, just have a minor comment about the CTA's container element when rendered in rich text.

I created #499 with a few changes that seem to resolve the TS errors. Basically I'm overriding the `callToActionDestination` field type so that it functions the same as our other rich text fields. I think it would be good to get feedback from @Benaiah on that one.

File: src/lib/components/ContentfulRichText/nodes/EmbeddedEntryBlock.svelte [hunk 3]
```diff
@@ -34,6 +35,13 @@
     height={entryBlock.linkedImage?.height ?? undefined}
     sizeType={getContext(imageSizeTypeKey)}
   />
+{:else if entryBlock?.__typename === "CallToAction"}
+  <span class="embedded-entry-CTA">
```
[the remark this answers is not in the document]
Comment (hinzed1127) [reply src/lib/components/ContentfulRichText/nodes/EmbeddedEntryBlock.svelte:39]: Yeah I agree the the juice isn't worth the squeeze for now. `EmbeddedEntryBlock` is more of a rich text edge case, in that it's conditionally rendering different types of content. I agree it's worth keeping an eye on if other rich text components end up needing similar functionality

**Review by Benaiah — APPROVED** [review]

**Pull request merged**